Sfapi1.7

hyunwoo18 opened the pull request:
Same code as sfapinew branch 
sfapi1.7 is branched off 1.7

mambelli reviewed (changes requested):
It seems that this PR includes changes unrelated to the SFAPI.
And they are substantial changes, not only formatting changes or other pre-commit caused changes
Eg. the AWS publisher has 2 more parameters in a function call
```
            end_point.send_dict(
...
                max_retries=self.max_retries,
                retry_interval=self.retry_interval,
```
This cannot be merged.
It seems the commits are many and are against master and not 1.7?
To change 1.7 you should start from 1.7 and cherry-pick or manually add the changes, doing merges will cause trouble

hyunwoo18 commented:
This is weird because the following is what I did:

[root@fermicloud827 sources]# git branch
  master
* sfapinew

[root@fermicloud827 sources]#  git checkout upstream/1.7

[root@fermicloud827 sources]# git branch
* (HEAD detached at upstream/1.7)
  master
  sfapinew

[root@fermicloud827 sources]# git checkout -b 1.7
Switched to a new branch '1.7'

[root@fermicloud827 sources]# git branch
* 1.7
  master
  sfapinew

[root@fermicloud827 sources]# git checkout -b sfapi1.7 1.7
Switched to a new branch 'sfapi1.7'
    
[root@fermicloud827 sources]# git branch
  1.7
  master
* sfapi1.7
  sfapinew

And here I only updated 2 files as can be seen from git diff command
[root@fermicloud827 sources]# git diff 1.7 sfapi1.7

diff --git a/setup.py b/setup.py
index 40279d4..d57b8c9 100644
--- a/setup.py
+++ b/setup.py
@@ -33,7 +33,11 @@ runtime_require = [
     "numpy == 1.19.5; python_version <= '3.6'",
     "numpy >= 1.19.5; python_version >= '3.7'",
     "pandas == 1.1.5; python_version <= '3.6'",
-    "pandas >= 1.1.5; python_version >= '3.7'"
+    "pandas >= 1.1.5; python_version >= '3.7'",
+    "authlib >= 1.2.1",
+    "cryptography >= 40.0.2",
+    "pem >= 21.2.0",
+    "jwt"
 ]

diff --git a/src/decisionengine_modules/NERSC/sources/NerscSFApi.py b/src/decisionengine_modules/NERSC/sources/NerscSFApi.py
new file mode 100644
index 0000000..2e2a0fe
--- /dev/null
+++ b/src/decisionengine_modules/NERSC/sources/NerscSFApi.py

pep8speaks commented:
Hello @hyunwoo18! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:





















* In the file [`src/decisionengine_modules/NERSC/sources/NerscSFApi.py`](https://github.com/HEPCloud/decisionengine_modules/blob/dfdca1ee2c657b91560d6f416e9251ff192ef027/src/decisionengine_modules/NERSC/sources/NerscSFApi.py):

> [Line 65:1](https://github.com/HEPCloud/decisionengine_modules/blob/dfdca1ee2c657b91560d6f416e9251ff192ef027/src/decisionengine_modules/NERSC/sources/NerscSFApi.py#L65): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace

hyunwoo18 commented:
I opened a new PR 470 to replace this.
Closing this one.
